### PR TITLE
fix: query&display metadata with new metadata struct

### PR DESCRIPTION
**Pull request by tyyzqmf**
----

*By submitting this pull request, I confirm that my contribution is made under the terms of the Apache-2.0 license*

## Summary

(describe what this merge request does)

## Implementation highlights

(describe how the merge request does for feature changes, share the RFC link if it has)

## Test checklist

- [ ] add new test cases
- [x] all code changes are covered by unit tests
- [ ] end-to-end tests
  - [x] deploy web console with CloudFront + S3 + API gateway
  - [ ] deploy web console within VPC
  - [ ] deploy ingestion server
    - [ ] with MSK sink
    - [ ] with KDS sink
    - [ ] with S3 sink
  - [ ] deploy data processing
  - [ ] deploy data modeling
    - [ ] new Redshift Serverless
    - [ ] provisioned Redshift
    - [ ] Athena
  - [ ] deploy with reporting

## Is it a breaking change

- [ ] add parameters without default value in stack
- [ ] introduce new service permission in stack
- [ ] introduce new top level stack module

## Miscellaneous

- [ ] introduce new symbol link source file(s) to be shared among infra code, web console frontend, and web console backend

### CODE REVIEW

File: src/control-plane/backend/lambda/api/model/metadata.ts [hunk 1]
```diff
@@ -60,8 +60,10 @@ export interface IMetadataEventParameter {
   platform: MetadataPlatform[];
   readonly parameterType: MetadataParameterType;
   readonly valueType: MetadataValueType;
-  valueEnum: IMetadataAttributeValue[];
+  valueEnum: string[];
```
Comment (zxkane) [inline src/control-plane/backend/lambda/api/model/metadata.ts:63]: Let's remove the values and keep the array of `IMetadataAttributeValue` only.

**Review by zxkane — CHANGES_REQUESTED** [review]

File: src/control-plane/backend/lambda/api/model/metadata.ts (hunk 1 above)
Comment (tyyzqmf) [reply src/control-plane/backend/lambda/api/model/metadata.ts]: removed from response

File: src/control-plane/backend/lambda/api/model/metadata.ts [hunk 2]
```diff
@@ -60,8 +60,10 @@ export interface IMetadataEventParameter {
   platform: MetadataPlatform[];
   readonly parameterType: MetadataParameterType;
   readonly valueType: MetadataValueType;
-  valueEnum: IMetadataAttributeValue[];
+  valueEnum?: string[];
```
Comment (zxkane) [inline src/control-plane/backend/lambda/api/model/metadata.ts:63]: Is it still useful to keep it?

File: src/control-plane/backend/lambda/api/model/metadata.ts [hunk 3]
```diff
@@ -77,14 +79,18 @@ export interface IMetadataUserAttribute {
...
+
   readonly attributeId: string;
   readonly name: string;
   displayName: string;
   description: string;
   readonly metadataSource: MetadataSource;
   hasData: boolean;
   readonly valueType: MetadataValueType;
-  valueEnum: IMetadataAttributeValue[];
+  valueEnum?: string[];
```
Comment (zxkane) [inline src/control-plane/backend/lambda/api/model/metadata.ts:91]: Is it still useful to keep it?

**Review by zxkane — CHANGES_REQUESTED** [review]

File: src/control-plane/backend/lambda/api/model/metadata.ts (hunk 2 above)
Comment (tyyzqmf) [reply src/control-plane/backend/lambda/api/model/metadata.ts:63]: we need read it from DDB.

File: src/control-plane/backend/lambda/api/model/metadata.ts (hunk 3 above)
Comment (tyyzqmf) [reply src/control-plane/backend/lambda/api/model/metadata.ts:91]: we need read it from DDB and then patch display info to values.

**Review by zxkane — APPROVED** [review]

**Pull request merged**